### PR TITLE
Refactor telemetry export types to records

### PR DESCRIPTION
**Pull request by onurkaracali**
## Summary

Follow-up to #36807 — addresses review feedback from @hakonhall and @hmusum.

- **Convert to Java records**: `TelemetryExportConfig`, `Exporter`, and `Auth` are now records instead of classes, reducing boilerplate (`equals`/`hashCode`/`toString` auto-generated)
- **Remove config-model duplicates**: Deleted `TelemetryAuth`, `TelemetryExport`, and `TelemetryExporter` from config-model. `Admin.java` and `DomAdminBuilderBase.java` now use the canonical types from config-provisioning directly
- **Add factory methods on `Auth`**: `bearerToken()`, `apiKey()`, `basicAuth()` — each validates its required fields and constructs the record with the correct `type` string
- **Add `ExporterType` enum on `Exporter`**: Replaces raw string type with `otlp`, `otlphttp`, `googlecloud`
- **Rename**: `TelemetryExportConfig` → `TelemetryExportConfiguration` (and serializer accordingly)

### Before (config-model had its own parallel types)

```
config-model/admin/telemetry/TelemetryAuth.java      ← deleted
config-model/admin/telemetry/TelemetryExport.java     ← deleted
config-model/admin/telemetry/TelemetryExporter.java   ← deleted
config-provisioning/TelemetryExportConfig.java         ← was a class
```

### After (single source of truth in config-provisioning)

```
config-provisioning/TelemetryExportConfiguration.java  ← record
  ├── Exporter (record, with ExporterType enum)
  └── Auth (record, with factory methods)
```

`DomAdminBuilderBase` imports config-provisioning types directly and constructs them during XML parsing. No intermediate config-model types needed.

## Test plan

- [x] `TelemetryExportConfigurationSerializerTest` — 17 tests (round-trip, JSON parse/serialize, equality)
- [x] `TelemetryExportTest` — 14 tests (XML parsing, all auth types, validation errors, edge cases)

### CODE REVIEW

**Review by hmusum — DISMISSED** [review]
Nice!

File: config-provisioning/src/main/java/com/yahoo/config/provision/TelemetryExportConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Immutable telemetry export configuration for an application deployment.
+ *
+ * @author onur
+ */
+public record TelemetryExportConfiguration(List<Exporter> exporters) {
```
Comment (hakonhall) [inline config-provisioning/src/main/java/com/yahoo/config/provision/TelemetryExportConfiguration.java]: →...Export*er*... ?

**Review by hakonhall — DISMISSED** [review]

File: config-provisioning/src/main/java/com/yahoo/config/provision/TelemetryExportConfiguration.java (hunk 1 above)
Comment (hmusum) [reply config-provisioning/src/main/java/com/yahoo/config/provision/TelemetryExportConfiguration.java]: Yeah, fixed

**Review by hmusum — APPROVED** [review]
